plugins/websocket_server/WebsocketServer.cc

```diff
@@ -998,6 +998,65 @@ void WebsocketServer::OnMessage(int _socketId, const std::string _msg)
   {
     this->OnAsset(_socketId, frameParts);
   }
+  else if (frameParts[0] == "sim")
+  {
+    std::string topic = frameParts[1];
+    std::string action = frameParts[2];
+
+    // Terminate simulation
+    if (action == "stop")
+    {
+      igndbg << "Stopping simulation per request via websocket\n";
+
+      // Default topic if not supplied
+      if (topic.empty())
+      {
+        topic = "/server_control";
+      }
+
+      ignition::msgs::ServerControl req;
+      req.set_stop(true);
+      ignition::msgs::Boolean rep;
+
+      bool result;
+      unsigned int timeout = 2000;
```

Comment:
What units is the timeout in?  Milliseconds?

If so, a 2 second timeout here is a really long time.  I guess in this case we know that we are trying to stop the simulation, but blocking for 2 seconds seems long.  Maybe we should make this call asynchronous?

Same goes for the call below.

Reply:
So I used the 2000 because all the other `timeout`s in this file are 2000. The other ones are things like requesting world info, scene info, and particle effect. I don't know the context for choosing 2000 - world info doesn't take too long, but maybe it is waiting for things to boot up completely if the request is received in the very beginning? I think those requests shouldn't take longer than shutting down, so I kept the timeout for shutting down the same. Another reason is that I don't know whether shutting down should be shorter or longer than 2000, so I'm not sure which direction to go.